front/logger/withlogging.ts

```diff
@@ -167,29 +168,71 @@ export function withLogging<T>(
   };
 }
 
-export function dustErrorToApiError(
+// TODO: add other error codes as we use this method in other places
+type SupportedDustErrorType = Omit<DustError, "code"> & {
+  code:
+    | "unauthorized"
```

Comment:
Meta: if we change the code in DustError this will fail to pick it up right? I think we should just switch on the code and have a default case instead of an assertNever in that situation.

But I think you're already making a pretty deep design decision about error handling here which I am personally not yet aligned with.

Reply:
Yes :+1: